compliance/lib/data_adapter/data_adapter.rego

```diff
@@ -43,16 +42,26 @@ process_name = name {
 
 process_args_list = args_list {
 	is_process
-	args_list := split(input.resource.command, " ")
+
+	# Gets all the process arguments of the current process
+	# Expects format as the following: --<key><delimiter><value> for example: --config=a.json
+	# Notice that the first argument is always the process path
+	args_list := split(input.resource.command, " --")
 }
 
-process_args = args {
-	args := {arg: value | [arg, value] = common.split_key_value(process_args_list[_])}
+# This method creates a process args object
+# The object will cotains all the process `flags` and their matching values as object key,value accordingly
+process_args(delimiter) = {arg_key: arg_value |
+	arguments = split(process_args_list[_], delimiter)
+	arg_key = concat("", ["--", arguments[0]])
+
+	# We would like to take the entire string after the first delimiter
+	arg_value = concat(delimiter, array.slice(arguments, 1, count(arguments) + 1))
 }
```

Comment:
```suggestion
# This method creates a process args object
# The object will cotains all the process `flags` and their matching values as object key,value accordingly
process_args(delimiter) = {flag: value | [flag, value] = parse_argument(process_args_list[_], delimiter)}

parse_argument(argument, delimiter) = [flag, value] {
	splitted_argument = split(argument, delimiter)
	flag = concat("", ["--", splitted_argument[0]])

	# We would like to take the entire string after the first delimiter
	value = concat(delimiter, array.slice(splitted_argument, 1, count(splitted_argument) + 1))
}
```

I just hate the way OPA formats this, so I suggest splitting it into two functions. Also  `arguments = split(process_args_list[_], delimiter)` is not accurate, see my suggestion. 
let me know what you think.

Reply:
I find both options readable.
I don't find any option more readable than the other, I find the first one a bit more elegant.

If you find it unreadable, or complex I can change it.

WDT?

Reply:
I would prefer we change it